data/modules/SearchRescue/SearchRescue.lua

```diff
@@ -1813,6 +1813,10 @@ local onEnterSystem = function (player)
 	 mission.target = createTargetShip(mission)
       end
    end
+	  
+      for _,discarded_ship in pairs(discarded_ships) do
```

Comment:
Not sure why you are emptying "discarded_ships" here too? Shouldn't the insertion within "onLeaveSystem" (see line 1835 below) take care of it? Or am I missing a situation where you are not leaving a system but entering another?